From d01a8ea024c7f87d00f38aaf1364a5c7097f94c3 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 25 May 2022 06:02:14 +0300
Subject: [PATCH] [SW-0] rom: update rr config

As per the updated security specifications, the usage of RR bit in
efuse is to enable/disable rr config done in hardware, and not for
rom as a backup for rr config failure in hardware.

Hence this rr config can be removed from rom.

Change-Id: Icb9baf3c7b6d6f7f81af51e59ae3a67bebdf56c4
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 327a213..41617aa 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -509,36 +509,10 @@
 {
 	security_i2c_axprot_config();
 	security_smi_axprot_config();
-
-	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
-	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
-
-	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
-	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
-
-	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
-
-	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
 }
 
 void security_disable_twi_if(void)
 {
-	if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RR_MASK) == 0)
-		return;
-
-	rr_ctx_lbw[RR_BLOCK_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
-	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
-
-	rr_ctx_lbw[RR_BLOCK_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
-	security_rr_twi_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
-
-	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
-
-	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_twi_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
 }
 
 void security_handle_twi(void)
@@ -558,21 +532,6 @@
 void security_disable_jtag_if(void)
 {
 	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
-
-	if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RR_MASK) == 0)
-		return;
-
-	rr_ctx_lbw[RR_BLOCK_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
-	security_rr_jtag_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
-
-	rr_ctx_lbw[RR_BLOCK_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
-	security_rr_jtag_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
-
-	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_jtag_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
-
-	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_jtag_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
 }
 
 static void security_jtag_axprot_config(void)
@@ -590,19 +549,6 @@
 void security_enable_jtag_if(void)
 {
 	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
-
-	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
-	security_rr_jtag_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
-
-	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
-	security_rr_jtag_lbw_config(&rr_ctx_lbw[RR_OPEN_ALL]);
-
-	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_jtag_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
-
-	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
-	security_rr_jtag_hbw_config(&rr_ctx_hbw[RR_OPEN_ALL]);
-
 	security_jtag_axprot_config();
 }
 
@@ -689,35 +635,12 @@
 			arcboot_rom_global.config_fuse;
 }
 
-static void security_rr_lbw_config(void)
-{
-	rr_ctx_lbw[RR_BLOCK_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
-	security_rr_pcie_lbw_config(&rr_ctx_lbw[RR_BLOCK_ALL]);
-}
-
-static void security_rr_hbw_config(void)
-{
-	rr_ctx_hbw[RR_BLOCK_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
-	security_rr_pcie_hbw_config(&rr_ctx_hbw[RR_BLOCK_ALL]);
-}
-
-static void security_rr_config(void)
-{
-	if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_RR_MASK) == 0)
-		return;
-
-	security_rr_lbw_config();
-	security_rr_hbw_config();
-}
-
 void arcboot_rom_security_init(void)
 {
 	/* based on efuse, handle jtag enable/disable */
 	security_handle_jtag_if();
 	/* based on efuse, handle twi enable/disable */
 	security_handle_twi();
-
-	security_rr_config();
 
 	if (arcboot_rom_global.die_id)
 		return;
